Adyen/HttpClient/HttpURLConnectionClient.cs

```diff
@@ -197,6 +197,17 @@ public HttpWebRequest GetHttpWebRequest(string endpoint, Config config, bool isA
             {
                 httpWebRequest.Proxy = config.Proxy;
             }
+
+            if (!string.IsNullOrEmpty(config.CertificatePath))
+            {
+                using (X509Store store = new X509Store(StoreName.Root, StoreLocation.CurrentUser))
+                {
+                    var x509Certificate2 = new X509Certificate2(config.CertificatePath);
+                    store.Open(OpenFlags.ReadWrite);
+                    store.Add(x509Certificate2);
```

Comment:
I think we should have this code elsewhere. Several things to consider:

- Performance for opening/closing the store per HttpWebRequest call
- The GetHttpWebrequest-method should only be responsible for adding things to the HttpWebRequest itsself and not add certificates to a particular certificate store. What happens if this method is called again? config.CertificatePath will not be empty and we'll add the certificate again ;)

Reply:
I agree with the performance. The other calls works even we set a custom ca store with a specific certificate
Any ideas for the code location ?